### PR TITLE
Add a manual test case for clones ListViewItem to WinformsControlsApp

### PR DESCRIPTION
**Pull request by SimonZhao888**
https://github.com/dotnet/winforms/issues/11658#issuecomment-2246680860

Add a manual test case for clones ListViewItem to WinformsControlsApp.

### Description 

When we try to call the ListView's derived class to use a clone of ListViewItem, we need to be aware that the clone's ListView is actually null, so we need to add an extra judgment when calling methods like OnItemChecked
```
if (e.Item.ListView == this)
{
     ...
} 
```
to ensure that the code doesn't throw an exception at runtime because the ListView property of the cloned object is null.

###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/winforms/pull/11785)

### CODE REVIEW

**Review by LeafShi1 — COMMENTED** [review]
LGTM! Please add description and video to show your changes

File: src/System.Windows.Forms/tests/IntegrationTests/WinformsControlsTest/ListViewTest.cs [hunk 1]
```diff
@@ -263,4 +283,20 @@ private void btnReplaceImageListView1_Click(object sender, EventArgs e)
 
         listView1.Refresh();
     }
+
+    protected class SubListView : ListView
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms/tests/IntegrationTests/WinformsControlsTest/ListViewTest.cs:287]: Please rename this class to something that describes it's purpose. And move it to a new file

File: src/System.Windows.Forms/tests/IntegrationTests/WinformsControlsTest/ListViewTest.cs [hunk 2]
```diff
@@ -263,4 +283,20 @@ private void btnReplaceImageListView1_Click(object sender, EventArgs e)
...
+    {        
+        public void TryOnItemChecked(ItemCheckedEventArgs e)
+        {
+            if (!CheckBoxes)
+            {
+                return;
+            }
+
+            if(e.Item.ListView == this)
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms/tests/IntegrationTests/WinformsControlsTest/ListViewTest.cs:296]:
```suggestion
            if (e.Item.ListView == this)
```

File: src/System.Windows.Forms/tests/IntegrationTests/WinformsControlsTest/ListViewTest.cs [hunk 3]
```diff
@@ -35,6 +34,18 @@ public ListViewTest()
 
         AddCollapsibleGroupToListView();
         AddGroupTasks();
+
+        string[] TestItems = { "Item 1", "Item 2", "Item 3" };
+        listView3.RetrieveVirtualItem += (s, e) =>
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms/tests/IntegrationTests/WinformsControlsTest/ListViewTest.cs:39]: please reference the workitem this is related to in a comment.

File: src/System.Windows.Forms/tests/IntegrationTests/WinformsControlsTest/ListViewTest.cs [hunk 4]
```diff
@@ -196,6 +207,15 @@ private void listView2_Click(object sender, System.EventArgs e)
         MessageBox.Show(this, "listView2_Click", "event");
     }
 
+    private void listView3_Click(object sender, System.EventArgs e)
+    {
+        var item = ((System.Windows.Forms.ListView)sender).FocusedItem;
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms/tests/IntegrationTests/WinformsControlsTest/ListViewTest.cs:212]: Do you need the full namespace name here?

File: src/System.Windows.Forms/tests/IntegrationTests/WinformsControlsTest/ListViewTest.cs [hunk 5]
```diff
@@ -263,4 +283,20 @@ private void btnReplaceImageListView1_Click(object sender, EventArgs e)
 
         listView1.Refresh();
     }
+
+    protected class SubListView : ListView
+    {        
+        public void TryOnItemChecked(ItemCheckedEventArgs e)
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms/tests/IntegrationTests/WinformsControlsTest/ListViewTest.cs:289]:
```suggestion
        public void InvokeOnItemChecked(ItemCheckedEventArgs e)
```

Comment (Tanya-Solyanik) [reply src/System.Windows.Forms/tests/IntegrationTests/WinformsControlsTest/ListViewTest.cs]: @SimonZhao888 - FYI APIs whose names start with Try usually contain try/catch in the implementation. See Int32.TryParse vs Int32.Parse

**Review by Tanya-Solyanik — APPROVED** [review]
Thank you!

**Comment by Liv-Goh** [comment]
Verified this PR on latest WinForms repo from release/9.0 branch, it has been fixed: the code doesn't throw an exception at runtime when the ListView property of the cloned object is null.
WinFormsControlTestsListView

**Comment by Liv-Goh** [comment]
Verified this PR on latest WinForms repo from release/9.0-rc1 branch WinformsControlsApp, it has been fixed, the test result is same as above

**Pull request merged**